src/JSInterop/Microsoft.JSInterop/src/JSRuntimeExtensions.cs

```diff
@@ -45,6 +136,26 @@ public static async ValueTask InvokeVoidAsync(this IJSRuntime jsRuntime, string
         return jsRuntime.InvokeAsync<TValue>(identifier, args);
     }
 
+    /// <summary>
+    /// Invokes the specified JavaScript function asynchronously.
+    /// <para>
+    /// <see cref="JSRuntime"/> will apply timeouts to this operation based on the value configured in <see cref="JSRuntime.DefaultAsyncTimeout"/>. To dispatch a call with a different timeout, or no timeout,
+    /// consider using <see cref="IJSRuntime.InvokeAsync{TValue}(string, CancellationToken, object[])" />.
+    /// </para>
+    /// </summary>
+    /// <param name="jsRuntime">The <see cref="IJSRuntime"/>.</param>
+    /// <typeparam name="TValue">The JSON-serializable return type.</typeparam>
+    /// <param name="identifier">An identifier for the function to invoke. For example, the value <c>"someScope.someFunction"</c> will invoke the function <c>window.someScope.someFunction</c>.</param>
+    /// <param name="options">The <see cref="JsonSerializerOptions"/> to use for JSON serialization and deserialization.</param>
+    /// <param name="args">JSON-serializable arguments.</param>
+    /// <returns>An instance of <typeparamref name="TValue"/> obtained by JSON-deserializing the return value.</returns>
+    public static ValueTask<TValue> InvokeAsync<[DynamicallyAccessedMembers(JsonSerialized)] TValue>(this IJSRuntime jsRuntime, string identifier, JsonSerializerOptions options, params object?[]? args)
```

Comment:
Not strictly necessary, but you might want to consider exposing overloads accepting a `JsonTypeInfo<TValue>` instead of `JsonSerializerOptions`. That would make it in line with APIs in STJ and System.Net.Http.Json and remove the `DynamicallyAccessedMembers` restriction from `TValue`.